### PR TITLE
Google Analytics tag from Site Kit

### LINKED ISSUE #225: Google Analytics (Site Kit)
Automatic insertion of Google Analytics tracking code when Site Kit is installed

### PR DESCRIPTION
**Pull request by miina**
Fixes #225. 

Adds google analytics tag if it's set in Site Kit plugin and also adds an additional action to allow adding more analytics tags.

### CODE REVIEW

File: tests/phpunit/tests/Story_Renderer.php [hunk 1]
```diff
@@ -107,4 +107,51 @@ public function test_replace_body_end_tag() {
 
 		$this->assertSame( $expected, $actual );
 	}
+
+	public function test_maybe_add_analytics() {
```
Comment (spacedmonkey) [inline tests/phpunit/tests/Story_Renderer.php]: Nit pick. It is normally better to have this as two tests. One to test if you should and when you should not insert the tag.

File: includes/Story_Renderer.php [hunk 2]
```diff
@@ -133,6 +133,106 @@ protected function add_poster_images( $content ) {
...
+	protected function maybe_add_analytics( $content ) {
+		ob_start();
+
+		// @todo This would ideally be used in Site Kit plugin directly to reuse the method already existing there.
+		do_action( 'web_stories_print_analytics' );
+
+		$this->print_amp_analytics();
+
+		$output = (string) ob_get_clean();
+		if ( ! empty( $output ) ) {
```
Comment (spacedmonkey) [inline includes/Story_Renderer.php:271]: A comment here would be useful to explain that you need to add the amp anaytics script if analytics are loaded.

**Review by spacedmonkey — APPROVED** [review]
I have approved with two tiny nit picks. But otherwise, great work.

File: includes/Story_Renderer.php [hunk 3]
```diff
@@ -133,6 +133,106 @@ protected function add_poster_images( $content ) {
...
+	 * @return void
+	 */
+	public function print_analytics_script() {
+		?>
+		<script async="async" src="https://cdn.ampproject.org/v0/amp-analytics-0.1.js" custom-element="amp-analytics"></script>
+		<?php
+	}
+
+	/**
+	 * Prints AMP Analytics based on Site Kit configuration.
```
Comment (swissspidy) [inline includes/Story_Renderer.php:148]: Can you add a link here to where you copied this from? Makes it easier to update in the future.

Comment (miina) [reply includes/Story_Renderer.php:148]: Just updated it based on the configuration needed for Stories based on [this doc](https://docs.google.com/document/d/1q8rkYYMKZh3R2eqkpwgGkx6QX0oSXoLYtIO6ml1BFD0/edit#).

File: tests/phpunit/tests/Story_Renderer.php (hunk 1 above)
Comment (miina) [reply tests/phpunit/tests/Story_Renderer.php]: @spacedmonkey Tests are updated.

Note that the gtag config also got updated based on [this doc](https://docs.google.com/document/d/1q8rkYYMKZh3R2eqkpwgGkx6QX0oSXoLYtIO6ml1BFD0/edit#), would be great if you could re-review that part.

File: includes/Story_Renderer.php (hunk 3 above)
Comment (miina) [reply includes/Story_Renderer.php:148]: (Meaning that now the configuration is not copied from Site Kit anymore.)

**Review by swissspidy — APPROVED** [review]
Looks like a good first start. Seems like we'll have to refactor some bits later anyway for #570, in accordance with the integration epic.

**Pull request merged**